src/include/storage/storage_structure/in_mem_file.h

```diff
@@ -23,7 +27,7 @@ class InMemFile {
 
     std::string readString(common::ku_string_t* strInInMemOvfFile);
 
-    void flush();
+    void flush(common::FileSystem* fs);
```

Comment:
Instead of passing `fs` to the `flush` function, which requires `HashIndexBuilder` to hold a private field of `fs`, we can construct the necessary `fileInfo` inside the constructor of `InMemFile` and keep `fileInfo` as a private field of `InMemFile` which makes sense to me, and then we only need to pass `fs` to `InMemFile`'s constructor.

Reply:
done